server/sonar-server/src/main/java/org/sonar/server/rule/index/RuleIndexer.java

```diff
@@ -76,19 +85,86 @@ public void indexOnStartup(Set<IndexType> uninitializedIndexTypes) {
     bulk.stop();
   }
 
-  public void indexRuleDefinition(RuleKey ruleKey) {
-    indexRuleDefinitions(singletonList(ruleKey));
+  public void commitAndIndex(DbSession dbSession, RuleKey ruleKey) {
+    commitAndIndex(dbSession, singletonList(ruleKey));
   }
 
-  public void indexRuleDefinitions(List<RuleKey> ruleKeys) {
-    BulkIndexer bulk = new BulkIndexer(esClient, RuleIndexDefinition.INDEX, Size.REGULAR);
-    bulk.start();
+  public void commitAndIndex(DbSession dbSession, Collection<RuleKey> ruleDtos) {
+    List<EsQueueDto> items = ruleDtos.stream()
+      .map(key -> EsQueueDto.create(EsQueueDto.Type.RULE, key.toString()))
+      .collect(MoreCollectors.toArrayList());
+
+    dbClient.esQueueDao().insert(dbSession, items);
+    dbSession.commit();
+    postCommit(dbSession, ruleDtos, items);
+  }
+
+  public void commitAndIndex(DbSession dbSession, OrganizationDto organizationDto, RuleKey ruleKey) {
+    commitAndIndex(dbSession, organizationDto, singletonList(ruleKey));
+  }
+
+  public void commitAndIndex(DbSession dbSession, OrganizationDto organizationDto, Collection<RuleKey> ruleKeys) {
+    List<EsQueueDto> items = ruleKeys.stream()
+      .map(l -> EsQueueDto.create(EsQueueDto.Type.RULE, l.toString()))
+      .collect(MoreCollectors.toArrayList());
+
+    dbClient.esQueueDao().insert(dbSession, items);
+    dbSession.commit();
+    postCommit(dbSession, ruleKeys, items);
+  }
+
+  /**
+   * Entry point for Byteman tests. See directory tests/resilience.
+   * The parameter "ruleKeys" is used only by the Byteman script.
+   */
+  private void postCommit(DbSession dbSession, Collection<RuleKey> ruleKeys, Collection<EsQueueDto> items) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "ruleKeys". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1172)